midi_app_controller/state/state_manager.py

```diff
@@ -284,6 +284,34 @@ def start_handling(self) -> None:
             midi_out=self._midi_out,
         )
 
+    def is_recent_port_saved(self, port_name: str) -> bool:
+        """Checks if there is saved port for the controller.
+        port_name is either in or out
+        """
+        if self.selected_controller:
+            return port_name in self.recent_midi_ports_for_controller.get(
+                self.selected_controller.path, {}
+            )
+        return False
+
+    def select_default_controller(self):
+        """Selects default port for the controller"""
+        if self.selected_controller:
+            controller = Controller.load_from(self.selected_controller.path)
+            if (
+                controller.preferred_midi_in
+                and controller.preferred_midi_in in self.get_available_midi_in()
```

Comment:
It can be simplified. If `controller.preferred_midi_in` is `None` then it will not be in `self.get_available_midi_in()`.